osu.Game/Overlays/ChatOverlay.cs

```diff
@@ -303,6 +303,29 @@ private void currentChannelChanged(ValueChangedEvent<Channel> e)
                 channelManager.MarkChannelAsRead(e.NewValue);
         }
 
+        /// <summary>
+        /// Highlights a certain message in the specified channel.
+        /// </summary>
+        /// <param name="message">The message to highlight.</param>
+        public void HighlightMessage(Message message)
+        {
+            Channel targetChannel;
+
+            if (channelManager.CurrentChannel.Value.Id == message.ChannelId)
+                targetChannel = channelManager.CurrentChannel.Value;
+            else
+            {
+                targetChannel = channelManager.JoinedChannels.SingleOrDefault(c => c.Id == message.ChannelId);
+
+                if (targetChannel != null)
+                    channelManager.CurrentChannel.Value = targetChannel;
+                else
+                    targetChannel = channelManager.JoinChannel(channelManager.AvailableChannels.SingleOrDefault(c => c.Id == message.ChannelId), true);
```

Comment:
This can fail if the message's channel isn't found in `AvailableChannels` - `JoinChannel()` returns null if passed a null channel, and this will die on a nullref at the `HighlightedMessage` assignment.

I'd wager that this is going to fail when you open a PM channel, leave it, and then try to highlight the message in the now-closed PM channel, as PM channels are not in `AvailableChannels`.

Reply:
Good catch, I should probably just pass the channel from the `MessageNotification` instead of doing this.